src/svm/impls/binary/binary.c

```diff
@@ -1555,31 +1555,35 @@ PetscErrorCode SVMLoadTrainingDataset_Binary(SVM svm,PetscViewer v)
 {
   MPI_Comm  comm;
 
-  Mat       Xt_training;
-  Mat       Xt_biased;
+  Mat       X_training;
+  Mat       X_biased;
   Vec       y_training;
 
   PetscReal bias;
   PetscInt  mod;
 
   PetscFunctionBegin;
   TRY( PetscObjectGetComm((PetscObject) svm,&comm) );
-  TRY( MatCreate(comm,&Xt_training) );
-  TRY( PetscObjectSetName((PetscObject) Xt_training,"Xt_training") );
+  /* Create matrix of training samples */
+  TRY( MatCreate(comm,&X_training) );
+  TRY( PetscObjectSetName((PetscObject) X_training,"X_training") );
+  TRY( PetscObjectSetOptionsPrefix((PetscObject) X_training,"X_training_") );
```

Comment:
`DatasetLoad_Binary` overrides the names of the matrix and vector - but I think it's wrong

Reply:
The beyond idea is: setting matrix and vector names to X_training, y_training, respectively. Afterwards, loading matrix/vector couple from file (HDF5, PETSc binary). In these files, related objects are named X (group in case of HDF5) and y (dataset in case of HDF5). Then, I temporary change their names to X/y and, afterwards, set they back. Okay, I think the more clear way will be that I set the names of objects after loading from HDF5/PETSc binary files.  RIght?